GVRf/Framework/framework/src/main/jni/objects/mesh.cpp

```diff
@@ -208,8 +211,159 @@ void Mesh::bindVertexAttributes(GLuint programID)
     unSetVaoDirty();
 }
 
+void Mesh::createAttributeMapping(int programId,
+        int& totalStride, int& attrLen)
+{
+    dumpActiveAttribues(programId);
```

Comment:
Should dumpActiveAttribues be called after the -1 check? I'd imagine glGetProgramiv would fail for -1 and then numActiveAtributes can be anything. Not that something will crash but still.. While you are at it you could rename it to dumpActiveAttribu**t**es.